src/ui/public/es.js

```diff
@@ -9,38 +9,39 @@ import 'elasticsearch-browser';
 import _ from 'lodash';
 import uiModules from 'ui/modules';
 
-let es; // share the client amongst all apps
+const plugins = [function (Client, config) {
+  // esFactory automatically injects the AngularConnector to the config
+  // https://github.com/elastic/elasticsearch-js/blob/master/src/lib/connectors/angular.js
+  class CustomAngularConnector extends config.connectionClass {
+    request = _.wrap(this.request, function (request, params, cb) {
+      if (String(params.method).toUpperCase() === 'GET') {
+        params.query = _.defaults({ _: Date.now() }, params.query);
+      }
+
+      return request.call(this, params, cb);
+    });
+  }
+
+  config.connectionClass = CustomAngularConnector;
+}];
+
 uiModules
   .get('kibana', ['elasticsearch', 'kibana/config'])
-  .service('es', function (esFactory, esUrl, $q, esApiVersion, esRequestTimeout) {
-    if (es) return es;
-
-    es = esFactory({
+  .service('es', function (esFactory, esUrl, esApiVersion, esRequestTimeout) {
```

Comment:
+1 to comments

It's `es` to preserve backwards compatibility with the old implementation, unfortunately.  It is more complicated now, but Kibana has been more complicated in this way when xpack is in the mix for awhile since monitoring also supports an entirely separate cluster.  This change is just formalizing the concept.  The two are the same under the default configuration.

We need sufficient integration testing with tribe node to help us avoid making cluster-choice based mistakes.  That's in the follow up meta ticket to address after feature freeze: https://github.com/elastic/kibana/issues/9640.  CI builds should run all possible selenium tests against a tribe configuration to help verify behaviors here.